app/assets/javascripts/discourse/app/components/glimmer-topic-timeline.js

```diff
@@ -0,0 +1,207 @@
+import GlimmerComponent from "discourse/components/glimmer";
+import { tracked } from "@glimmer/tracking";
+import optionalService from "discourse/lib/optional-service";
+
+export default class GlimmerTopicTimeline extends GlimmerComponent {
+  @tracked intersectionObserver = null;
+  @tracked dockAt = null;
+  @tracked dockBottom = null;
+  @tracked enteredIndex = this.args.enteredIndex;
+
+  mobileView = this.site.mobileView;
+  adminTools = optionalService();
```

Comment:
The ember service injection system is already optional, so I don't think we need `optionalService()` any more. We should `import { inject as service } from "@ember/service";`, then 

```suggestion
@service adminTools;
```

(I'll look at adding a deprecation notice for `optionalService()` :eyes:)

Reply:
Actually, ignore me. The standard `service()` helper raises an error if the service doesn't exist… so we do need `optionalService()` 👌